x/stake/querier/queryable.go

```diff
@@ -283,6 +254,35 @@ func queryUnbondingDelegation(ctx sdk.Context, cdc *codec.Codec, req abci.Reques
 	return res, nil
 }
 
+func queryRedelegations(ctx sdk.Context, cdc *codec.Codec, req abci.RequestQuery, k keep.Keeper) (res []byte, err sdk.Error) {
+	var params QueryRedelegationParams
+
+	errRes := cdc.UnmarshalJSON(req.Data, &params)
+	if errRes != nil {
+		return []byte{}, sdk.ErrUnknownRequest(string(req.Data))
+	}
+
+	var redels []types.Redelegation
+
+	if !params.DelegatorAddr.Empty() && !params.SrcValidatorAddr.Empty() && !params.DstValidatorAddr.Empty() {
```

Comment:
Maybe a `switch` would be easier to read here.

Reply:
++

Reply:
++

Reply:
Not sure, it's three different options that are being checked in weird configs

```
if !option1 && !option2 && !option3

else if option1 && !option2 && option 3

else
```

How do I write that cleanly in a switch statement?  3 tiered switch statements?